cmake/external/googletest.cmake

```diff
@@ -23,11 +23,20 @@ set(GTEST_MAIN_LIBRARIES gtest_main)
 set(GTEST_BOTH_LIBRARIES ${GTEST_LIBRARIES} ${GTEST_MAIN_LIBRARIES})
 
 foreach(lib IN LISTS GTEST_BOTH_LIBRARIES)
-  list(APPEND GTEST_BUILD_BYPRODUCTS ${GTEST_INSTALL_DIR}/lib/lib${lib}.a)
+  if (MSVC)
+    if (CMAKE_BUILD_TYPE MATCHES Debug)
```

Comment:
I'm not a fan of this. This will work for the common case:

```
cmake -DCMAKE_BUILD_TYPE=Debug ..
cmake --build .
```

However, MSVC targets also allow you to specify the target at build time:

```
cmake ..
cmake --build . --config Debug # this will fail looking for gtest.lib when only gtestd.lib exists
```

Should I open another issue? IMO, a more permanent solution would be https://crascit.com/2015/07/25/cmake-gtest/. This is also called out on the gtest README proper.

Reply:
which part of https://crascit.com/2015/07/25/cmake-gtest/ are you referring to?

Reply:
The post describes creating a very simple intermediate CMakeLists.txt.in which calls ExternalProject_Add on gtest; then the main CMakeLists.txt runs:

```cmake
# Download and unpack googletest at configure time
configure_file(CMakeLists.txt.in googletest-download/CMakeLists.txt)
execute_process(COMMAND "${CMAKE_COMMAND}" -G "${CMAKE_GENERATOR}" .
    WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/googletest-download" )
execute_process(COMMAND "${CMAKE_COMMAND}" --build .
    WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/googletest-download" )

...

# Add googletest directly to our build. This adds
# the following targets: gtest, gtest_main, gmock
# and gmock_main
add_subdirectory("${CMAKE_BINARY_DIR}/googletest-src"
                 "${CMAKE_BINARY_DIR}/googletest-build")
```

According to the post, the gtest_main and gtest targets are now directly available, so we don't have to reference build artifacts by name out of the gtest build dir, which is architecture+toolchain dependant. This is a little hacky, but it's also documented in the [googletest](https://github.com/google/googletest/tree/master/googletest#incorporating-into-an-existing-cmake-project) repo as well.

Reply:
lgtm